apps/authoring-services/src/main/java/com/infosys/lexauthoringservices/serviceimpl/CompetencyServiceImp.java

```diff
@@ -0,0 +1,146 @@
+        producer.push(competencyUpdateTopic, competency);
+    }
+
+    /**
+     * Update the copetency in neo4j & elasticsearch
+     *
+     * @param competency
+     * @return
+     */
+    @Override
+    public Competency processUpdateCompetencyData(Competency competency) {
+        StringBuilder builder = new StringBuilder();
+        String path = competencySearchPath.replace("{org}", competency.getRootOrg()).replace("{id}", competency.getId());
+        builder.append(sbextServiceURL).append(path);
+        Object response = fetchResult(builder);
+        if (ObjectUtils.isEmpty(response))
+            return competency;
+        List<HashMap<String, Object>> listOfIds = (List<HashMap<String, Object>>) response;
+        if (!listOfIds.isEmpty()) {
+            List<String> successRecords = new ArrayList<>();
+            List<String> failedRecords = new ArrayList<>();
+            logger.info("Competency update started ..............");
+            for (HashMap<String, Object> map : listOfIds) {
+                String lex_id = (String) map.get(IDENTIFIER_CONST);
+                try {
+                    Map<String, Object> contentData = contentCrudService.getContentNode(competency.getRootOrg(), lex_id);
+                    Object competencyObject = contentData.get(COMPETENCIES_CONST);
+                    if (!ObjectUtils.isEmpty(competencyObject)) {
+                        List<HashMap<String, String>> oldCompetencyData = (List<HashMap<String, String>>) competencyObject;
+                        List<HashMap<String, String>> newCompetencyData = new ArrayList<>();
+                        HashMap<String, String> competencyMap = new HashMap<>();
+                        for (HashMap<String, String> data : oldCompetencyData) {
+                            if (data.get(ID_CONST).equals(competency.getId())) {
+                                competencyMap = new HashMap<>();
+                                competencyMap.put(ID_CONST, competency.getId());
+                                competencyMap.put(NAME_CONST, competency.getName());
+                                competencyMap.put(DESCRIPTION_CONST, competency.getDescription());
+                                newCompetencyData.add(competencyMap);
+                            } else {
+                                newCompetencyData.add(data);
+                            }
+                        }
+                        contentData.put(COMPETENCIES_CONST, newCompetencyData);
+                        contentCrudService.updateContentMetaNode(competency.getRootOrg(), "", lex_id, contentData);
+                        successRecords.add(lex_id);
+                    }
+                } catch (Exception ex) {
+                    failedRecords.add(lex_id);
+                    logger.error(ex.toString());
+                }
+            }
+            if(!successRecords.isEmpty())
+            logger.info("Success Records : " + successRecords.toString());
```

Comment:
avoid concatenation - should be -- logger.info("Success Records : {}", successRecords.toString());